datumaro/components/environment.py

```diff
@@ -240,11 +240,27 @@ def make_transform(self, name, *args, **kwargs):
     def is_format_known(self, name):
         return name in self.importers or name in self.extractors
 
-    def detect_dataset(self, path):
-        return detect_dataset_format(
-            (
-                (format_name, importer.detect)
-                for format_name, importer in self.importers.items.items()
-            ),
-            path,
-        )
+    def detect_dataset(self, path, rejection_callback=None, depth=1):
```

Comment:
```suggestion
    def detect_dataset(self, path: str, rejection_callback: Optional[RejectionCallback] = None, depth: int = 1) -> List[str]:
```

`RejectionCallback = Callable[[str, RejectionReason, str], None]`